common release management with changesets

GSemikozov opened the pull request:
common release management for all packages (currently - python + js)

samuelcolvin commented on .github/workflows/ci.yml:
```diff
@@ -154,7 +133,12 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ steps.generate-token.outputs.token }}
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          COLUMNS: 150
+          UV_PYTHON: 3.13
+          UV_FROZEN: "1"
```
I don't think you need this, it's set globally.

GSemikozov replied:
removed

samuelcolvin commented on packages/python/package.json:
```diff
@@ -0,0 +1,18 @@
+  "name": "genai-prices",
+  "version": "0.0.8",
+  "main": "index.js",
+  "type": "module",
```
Please add a description explaining why this field exists.

samuelcolvin commented on .github/workflows/ci.yml:
```diff
@@ -154,7 +133,12 @@ jobs:
-          publish: npm run release
+          publish: |
+            npm run changeset-version
+            cd packages/python && uv publish --trusted-publishing always
```
I don't think you're building the python package!

I think you need `uv build --package genai-prices` in there before publish?

samuelcolvin reviewed (changes requested):
This won't work:
* the `js` job should be call should be called something more sensible like `release`
* that job needs the `environment` set, like it was in `release-pypi`
* I think it'll need `permissions` as well
* it should surely have `needs: [check]`

GSemikozov commented:
> This won't work:
> 
> * the `js` job should be call should be called something more sensible like `release`
> * that job needs the `environment` set, like it was in `release-pypi`
> * I think it'll need `permissions` as well
> * it should surely have `needs: [check]`

@samuelcolvin :
- main flow is named `release`, `js` has been renamed to `test-js` for more clarity
- added `environment` and `permissions` the same as it was in `release-pypi`
- we have now `needs: [check, test-js]`
- added release.md

GSemikozov commented on .github/workflows/ci.yml:
```diff
@@ -94,67 +94,89 @@ jobs:
-  release-pypi:
-    needs: [check]
-    if: success() && startsWith(github.ref, 'refs/tags/')
+  test-js:
```
fixed

GSemikozov commented on .github/workflows/ci.yml:
```diff
@@ -94,67 +94,89 @@ jobs:
+    env:
+      CI: true
+    steps:
+      - name: Begin CI...
```
removed names from test-js steps

GSemikozov commented on .github/workflows/ci.yml:
```diff
@@ -105,56 +119,46 @@ jobs:
+        uses: samuelcolvin/check-python-version@v4.1
+        with:
+          version_file_path: packages/python/pyproject.toml
         env:
```
yes, thanks! fixed

GSemikozov commented on .github/workflows/ci.yml:
```diff
@@ -105,56 +119,46 @@ jobs:
+            uv build --package genai-prices
+            uv publish --trusted-publishing always
+
+      - name: Check Python version
```
removed
